packages/next/src/generators/init/utils/createCustomTestConfig.ts

```diff
@@ -0,0 +1,43 @@
+import { joinPathFragments, Tree, updateJson } from '@nrwl/devkit';
+
+function getTestConfig(projectSourceRoot: string) {
+    return {
+        executor: '@nrwl/jest:jest',
+        outputs: ['{workspaceRoot}/coverage/{projectRoot}'],
+        options: {
+            jestConfig: `${projectSourceRoot}/jest.config.ts`,
+            passWithNoTests: true,
+            collectCoverage: true,
+            coverageReporters: ['text', 'html'],
+            collectCoverageFrom: [
+                './**/*.{js,jsx,ts,tsx}',
+                './!**/.next/**',
+                './!**/*.d.ts',
+                './!**/*.config.*',
+                './!**/_app.*',
+            ],
+            codeCoverage: true,
+        },
+        configurations: {
+            ci: {
+                ci: true,
+            },
+        },
+    };
+}
+
+export default async function addCustomTestConfig(
+    tree: Tree,
+    projectSourceRoot: string,
+) {
+    updateJson(
+        tree,
+        joinPathFragments(projectSourceRoot, 'project.json'),
+        json => {
+            const updatedJson = json;
+            updatedJson.targets = updatedJson.targets ?? {};
+            updatedJson.targets.test = getTestConfig(projectSourceRoot);
```

Comment:
Should we merge the configs with the existing test executor options?

Reply:
Thats a good shout. Guess only really need to merge the CI coverage config with whats there already. Will update